src/onnx/parse_convolution.cpp

```diff
@@ -42,18 +43,201 @@ struct parse_convolution : op_parser<parse_convolution>
+        zp->eval().visit([&](auto z) {
+            all_zeros = std::all_of(
+                z.begin(), z.end(), [&](auto val) { return float_equal(val, symmetric_value); });
+        });
+        return all_zeros;
+    }
+
+    static auto
+    qparam_broadcast_op(instruction_ref qparam, std::vector<std::size_t> lens, std::size_t axis)
+    {
+        if(qparam->get_shape().scalar())
+        {
+            return migraphx::make_op("multibroadcast", {{"out_lens", lens}});
+        }
+        else
+        {
+            return migraphx::make_op("broadcast", {{"out_lens", lens}, {"axis", axis}});
+        }
+    }
+
+    static instruction_ref handle_quant_bias(const operation& op,
+                                             const instruction_ref& input,
+                                             const instruction_ref& x,
+                                             const instruction_ref& weights,
+                                             const instruction_ref& x_zp,
+                                             const instruction_ref& w_zp,
+                                             onnx_parser::node_info& info)
+    {
+        instruction_ref ret = input;
+        if(not is_symmetric_zero_point(x_zp))
+        {
+            auto out_zp_1 = info.add_common_op(op.name(), x_zp, weights);
+            ret           = info.add_common_op("sub", ret, out_zp_1);
+        }
+
+        if(not is_symmetric_zero_point(w_zp))
+        {
+            auto out_zp_2 = info.add_common_op(op.name(), x, w_zp);
+            ret           = info.add_common_op("sub", ret, out_zp_2);
+        }
+
+        if(not(is_symmetric_zero_point(x_zp)) and not(is_symmetric_zero_point(w_zp)))
+        {
+            auto x_zp_bc =
+                info.add_instruction(qparam_broadcast_op(x_zp, x->get_shape().lens(), 0), x_zp);
+            auto w_zp_bc = info.add_instruction(
+                qparam_broadcast_op(w_zp, weights->get_shape().lens(), 0), w_zp);
+
+            auto out_zp_3 = info.add_instruction(op, x_zp_bc, w_zp_bc);
+
+            ret = info.add_common_op("add", ret, out_zp_3);
+        }
+        return ret;
+    }
+
+    static void handle_quant_inputs(const bool is_quant_conv,
+                                    instruction_ref& input,
+                                    instruction_ref& weights,
+                                    instruction_ref& input_zp,
+                                    instruction_ref& weight_zp,
+                                    onnx_parser::node_info& info)
+    {
+        if(not is_quant_conv)
+            return;
+
+        auto input_type  = input->get_shape().type();
+        auto weight_type = weights->get_shape().type();
+
+        // Handle uint8 bias and input shifts
+        instruction_ref offset_op;
+        if(((input_type == migraphx::shape::uint8_type) or
+            (weight_type == migraphx::shape::uint8_type)))
+        {
+            offset_op = info.add_literal(
```

Comment:
Is the offset_op always -128? Would it not depend upon the inputted zero_point of the input tensor (the zero point of uint8 can be anywhere between 0 to 255.. right?)

Reply:
Yes since we're moving everything from uint8 to int8 we need to subtract 128

Reply:
Thanks.  The correct functional boundary for this constant adjustment: it should be contained inside add_int8_shift().